postmortems/2021-07-27_update_ris3.md

```diff
@@ -0,0 +1,67 @@
+- Which file do I update? Do I update both of them? 
+- `ris3_views` assumes there is one common version datestamp across all tilesets. If I want to add a different version for the tileset for my new storypoint, there is no easy way of doing this short of including the specific tileset version on *every* storypoint.
+- `ris3_points.ttl` only contains data for storypoints. Are story objects necessary, and if so how and where do I add one? In `ris3_views`?
+- What happens if I run `ris3_views` at the same time as `--add-graph ris3_points.ttl`, which would add both of them to the model?
+
+### Update attempt 2
+
+After some investigation of the data in `ris3_views.py` and `ris3_points.ttl` I decided that attempting to the run them together was probably not a good idea. So I came up with a different approach: while the source data in `ris3_views.py` is out of date, it should still work for generating new triples for the new tileset. These can then be displayed as part of a turtle-format graph, and the output for the specific storypoint that I want to add can be copied into `ris3_points.ttl` which can then be added to the existing model.
+
+However, this approach also quickly proved problematic:
+- The storypoints in `ris3_points.ttl` have an `asOrdinal` property that is, as far as I can tell, completely undocumented. (It certainly doesn't appear anywhere in `sm_platform`.) It appears to be a simple incrementing counter, but since I was inserting new data into the sequence by adding my new storypoint there was the potential that I might overlap with existing `asOrdinal` values that were already encoded in `ris3_points.ttl`.
+- Running sheet-to-triples with just the `--add-graph ris3_points.ttl` argument and no transforms means it doesn't write anything to file -- a transform needs to be passed to do this. 
+- I hacked the run file to force it to always write to file, but the verbose output indicated that it was dropping far fewer triples as duplicates than expected (it should have dropped the entire file except for my new storypoint). Further investigation revealed that **the model pulled from staging was pointed at tilesets that were more recent than those versioned in `ris3_points.ttl`**.
+
+This last point meant that updating RIS3 using any code-based approach was impossible, because doing so relied on a versioned `ris3_points.ttl` that was out-of-date compared to whatever unversioned `ris3_points.ttl` had been used to generate the existing model data. Using the older copy would have undone newer changes with no way to recover them.
+
+### Update attempt 3
+
+The storypoint was eventually created using the following incredibly brittle method:
+
+- Add the new storypoint into `ris3_views.py` and run this transform only to generate the triples for the storypoint.
+- Open the triples json and find and copy only the new triples.

```

Comment:
The easy alternative here would be just take current content, and `--add-graph` one new file with the added story and story point. Don't need to do any json wrangling.